code/_onclick/hud/new_player.dm

```diff
@@ -0,0 +1,281 @@
+
+/atom/movable/screen/lobby/button/join/Click(location, control, params)
+	. = ..()
+	if(!.)
+		return
+	if(!SSticker?.IsRoundInProgress())
+		to_chat(hud.mymob, span_boldwarning("The round is either not ready, or has already finished..."))
+		return
+
+	//Determines Relevent Population Cap
+	var/relevant_cap
+	var/hpc = CONFIG_GET(number/hard_popcap)
+	var/epc = CONFIG_GET(number/extreme_popcap)
+	if(hpc && epc)
+		relevant_cap = min(hpc, epc)
+	else
+		relevant_cap = max(hpc, epc)
+
+	var/mob/dead/new_player/new_player = hud.mymob
+
+	if(SSticker.queued_players.len || (relevant_cap && living_player_count() >= relevant_cap && !(ckey(new_player.key) in GLOB.admin_datums)))
+		to_chat(new_player, span_danger("[CONFIG_GET(string/hard_popcap_message)]"))
+
+		var/queue_position = SSticker.queued_players.Find(new_player)
+		if(queue_position == 1)
+			to_chat(new_player, span_notice("You are next in line to join the game. You will be notified when a slot opens up."))
+		else if(queue_position)
+			to_chat(new_player, span_notice("There are [queue_position-1] players in front of you in the queue to join the game."))
+		else
+			SSticker.queued_players += new_player
+			to_chat(new_player, span_notice("You have been added to the queue to join the game. Your position in queue is [SSticker.queued_players.len]."))
+		return
+	new_player.LateChoices()
+
+/atom/movable/screen/lobby/button/join/proc/show_join_button(status)
+	SIGNAL_HANDLER
+	set_button_status(TRUE)
+	UnregisterSignal(SSticker, COMSIG_TICKER_ENTER_SETTING_UP)
+
+/atom/movable/screen/lobby/button/observe
+	screen_loc = "TOP:-40,CENTER:-54"
+	icon = 'icons/hud/lobby/observe.dmi'
+	icon_state = "observe_disabled"
+	base_icon_state = "observe"
+	enabled = FALSE
+
+/atom/movable/screen/lobby/button/observe/Initialize(mapload)
+	. = ..()
+	if(SSticker.current_state > GAME_STATE_STARTUP)
+		set_button_status(TRUE)
+	else
+		RegisterSignal(SSticker, COMSIG_TICKER_ENTER_PREGAME, .proc/enable_observing)
+
+/atom/movable/screen/lobby/button/observe/Click(location, control, params)
+	. = ..()
+	if(!.)
+		return
+	var/mob/dead/new_player/new_player = hud.mymob
+	new_player.make_me_an_observer()
+
+/atom/movable/screen/lobby/button/observe/proc/enable_observing()
+	SIGNAL_HANDLER
+	flick("[base_icon_state]_enabled", src)
+	set_button_status(TRUE)
+	UnregisterSignal(SSticker, COMSIG_TICKER_ENTER_PREGAME, .proc/enable_observing)
+
+/atom/movable/screen/lobby/button/changelog_button
+	icon = 'icons/hud/lobby/bottom_buttons.dmi'
+	icon_state = "changelog"
+	base_icon_state = "changelog"
+	screen_loc ="TOP:-122,CENTER:+2"
+
+/atom/movable/screen/lobby/button/changelog_button/Click(location, control, params)
+	. = ..()
+	usr.client?.changelog()
+
+/atom/movable/screen/lobby/button/poll
+	icon = 'icons/hud/lobby/bottom_buttons.dmi'
+	icon_state = "changelog"
+	base_icon_state = "changelog"
+	screen_loc = "TOP:-122,CENTER:-26"
+
+	var/new_poll = FALSE
```

Comment:
This doesn't appear to be used anymore. Can you add a little notification icon or something?

Reply:
This has already been sprited.

![bild](https://user-images.githubusercontent.com/49783092/130332212-29a5960c-c5ee-4412-bcaf-2b0bf86bcd52.png)

Reply:
yeah didnt implement this yet